src/atc/reporting/SlackNotifier.py

```diff
@@ -0,0 +1,93 @@
+import sys
+import time
+import traceback
+from datetime import datetime
+
+import requests
+
+from ..atc_exceptions import NoRunId
+from .JobReflection import JobReflection
+
+
+class SlackNotifier:
+    def __init__(self, webhookurl: str):
+        self.webhookurl = webhookurl
+
+    def notify_info(self, message: str = None, _stack_skip=1):
+        called_from = "".join(traceback.format_stack()[:-_stack_skip])
+        try:
+            job_name = JobReflection.get_job_name()
+            job_url = JobReflection.get_job_results_url()
+        except NoRunId:
+            job_url = job_name = ""
+
+        if job_name:
+            text = f"*A message was sent from your job {job_name}*\n"
+        else:
+            text = "*A message was sent from databricks*\n"
+
+        text += (
+            "\nSent at <!date^"
+            + str(int(time.time()))
+            + "^{date_long_pretty} {time_secs}|"
+            + str(datetime.now())
+            + ">\n"
+        )
+
+        if message:
+            text += f"\n{message}\n"
+
+        text += f"\nCaller info:\n```\n{called_from}\n```\n"
+
+        if job_url:
+            text += f"<{job_url}|Go To Job Results>\n"
+
+        # send the whole thing off to the webhook
+        res = requests.post(url=self.webhookurl, json={"text": text})
+        if res.status_code != 200:
+            # not raising an exception here makes this function more safe to call
+            print(
+                "WARNING: Slack Notifier received "
+                f"unexpected status code {res.status_code}"
+            )
+
+    def notify_exc(self):
+        if sys.exc_info() == (None, None, None):
+            return self.notify_info(
+                message="SlackNotifier.notify_exc()"
+                " was called with no active exception.",
+                _stack_skip=2,
+            )
+
+        try:
```

Comment:
Isnt line 62-93 the exact same as "notify_info"?

Reply:
I can see that text states something about "an exeption has occurred" and "message was sent". Could you make a notify-function that switch on notification type? Some thing like:
```python
def notify(text_type: str):
  assert text_type in {"notify","exeption"}
  try:
            job_name = JobReflection.get_job_name()
            job_url = JobReflection.get_job_results_url()
        except NoRunId:
            job_url = job_name = ""
        
        # MAKE SWITCH STATEMENT HERE REGARDING INFO/EXEPTION TYPE
        if job_name:
            text = f"*A message was sent from your job {job_name}*\n"
        else:
            text = "*A message was sent from databricks*\n"

       # MAKE SWITCH STATEMENT HERE REGARDING INFO/EXEPTION TYPE
       # Here is an example
        if text_type == "notify"

          text += (
              "\nSent at <!date^"
              + str(int(time.time()))
              + "^{date_long_pretty} {time_secs}|"
              + str(datetime.now())
              + ">\n"
          )
          
        if text_type == "exception"

          text += (
              "\nThe error occurred at <!date^"
              + str(int(time.time()))
              + "^{date_long_pretty} {time_secs}|"
              + str(datetime.now())
              + ">\n"
          )

        if message:
            text += f"\n{message}\n"

        text += f"\nCaller info:\n```\n{called_from}\n```\n"

        if job_url:
            text += f"<{job_url}|Go To Job Results>\n"

        # send the whole thing off to the webhook
        res = requests.post(url=self.webhookurl, json={"text": text})
        if res.status_code != 200:
            # not raising an exception here makes this function more safe to call
            print(
                "WARNING: Slack Notifier received "
                f"unexpected status code {res.status_code}"
            )
```